base/base64_body.hpp

```diff
@@ -0,0 +1,80 @@
+
+#pragma once
+
+#include "base/base64.hpp"
+
+#include <string>
+
+#include "absl/strings/escaping.h"
+
+namespace principia {
+namespace base {
+namespace internal_base64 {
+
+constexpr std::int64_t bits_per_byte = 8;
+constexpr std::int64_t bits_per_char = 6;
+
+template<bool null_terminated>
+void principia::base::internal_base64::Base64Encoder<null_terminated>::Encode(
+    Array<std::uint8_t const> input,
+    Array<char> output) {
+  std::string_view const input_view(reinterpret_cast<const char*>(input.data),
```

Comment:
Do we have string_view on mac?

Reply:
Non lo so.  There is an `absl::string_view`.  We'll switch to that if Linux/macOS doesn't have the `std` version.

Reply:
We can always summon the most boring kind of UB by
```C++
#if OS_MACOSX
namespace std {
using absl::string_view;
}  // namespace std
#endif
```